tests/zkevm/test_worst_bytecode.py

```diff
@@ -144,16 +156,27 @@ def generate_deploy_tx(contracts_to_deploy: int):
     post = {}
     deployed_contract_addresses = []
     for i in range(total_contracts_to_deploy):
-        deployed_contract_address = compute_create_address(
+        deployed_contract_address = compute_create2_address(
             address=factory_address,
-            nonce=i + 1,
+            salt=i,
+            initcode=initcode,
         )
         post[deployed_contract_address] = Account(nonce=1)
         deployed_contract_addresses.append(deployed_contract_address)
 
     opcode_code = (
-        sum(Op.POP(opcode(address=address)) for address in deployed_contract_addresses) + Op.STOP
+        # Setup memory for later CREATE2 address generation loop.
+        Op.MSTORE(0, factory_address)
+        + Op.MSTORE8(32 - 20 - 1, 0xFF)  # 0xFF prefix byte
+        + Op.MSTORE(32, 0)
+        + Op.MSTORE(64, initcode.keccak256())
```

Comment:
Only once we setup the `0xFF+address+salt+hash(initcode)`. The loop will reuse this memory region for generating the CREATE2 addresses.